rdt/hyper_transformer.py

```diff
@@ -458,15 +468,19 @@ def fit(self, data):
             data (pandas.DataFrame):
                 Data to fit the transformers to.
         """
+        self._unfit()
         self._input_columns = list(data.columns)
         self._populate_field_sdtypes(data)
 
         # Loop through field_transformers that are first level
-        for field in self.field_transformers:
+        for field in self.provided_field_transformers:
```

Comment:
If we use the `_learn_config` method described above then we can just loop through the combined dictionaries here by calling `_get_field_transformers()`